app/api/items/update_item.feature

```diff
@@ -0,0 +1,186 @@
+      """
+      {
+        "type": "Course",
+        "full_screen": ""
+      }
+      """
+    Then the response should be "updated"
+    And the table "items" at ID "50" should be:
+      | ID | sType  | sUrl | idDefaultLanguage | bTeamsEditable | bNoScore | sTextID | bTitleBarVisible | bCustomChapter | bDisplayDetailsInParent | bUsesAPI | bReadOnly | sFullScreen | bShowDifficulty | bShowSource | bHintsAllowed | bFixedRanks | sValidationType | iValidationMin | idItemUnlocked | iScoreMinUnlock | sTeamMode | bTeamsEditable | idTeamInGroup | iTeamMaxMembers | bHasAttempts | sAccessOpenDate      | sDuration | sEndContestDate      | bShowUserInfos | sContestPhase | iLevel | bNoScore | groupCodeEnter |
+      | 50 | Course | null |                 1 |              0 |        0 | null    | 1                | 0              | 0                       | 1        | 0         |             | 0               | 0           | 0             | 0           | All             | null           | null           | 100             | null      | 0              | null          | 0               | 0            | null                 | null      | null                 | 0              | Running       | null   | 0        | 0              |
+    And the table "items_strings" should stay unchanged
+    And the table "items_items" should be:
+      | idItemParent | idItemChild |
+      | 21           | 60          |
+    And the table "items_ancestors" should be:
+      | idItemAncestor | idItemChild |
+      | 21             | 60          |
+    And the table "groups_items" should stay unchanged
+
+  Scenario: Valid without any fields
+    Given I am the user with ID "1"
+    And the time now is "2018-01-01T00:00:00Z"
+    When I send a PUT request to "/items/50" with the following body:
+    """
+    {
+    }
+    """
+    Then the response should be "updated"
+    And the table "items" at ID "50" should be:
+      | ID | sType  | sUrl | idDefaultLanguage | bTeamsEditable | bNoScore | sTextID | bTitleBarVisible | bCustomChapter | bDisplayDetailsInParent | bUsesAPI | bReadOnly | sFullScreen | bShowDifficulty | bShowSource | bHintsAllowed | bFixedRanks | sValidationType | iValidationMin | idItemUnlocked | iScoreMinUnlock | sTeamMode | bTeamsEditable | idTeamInGroup | iTeamMaxMembers | bHasAttempts | sAccessOpenDate      | sDuration | sEndContestDate      | bShowUserInfos | sContestPhase | iLevel | bNoScore | groupCodeEnter |
+      | 50 | Root   | null | 1                 | 0              | 0        | null    | 1                | 0              | 0                       | 1        | 0         | default     | 0               | 0           | 0             | 0           | All             | null           | null           | 100             | null      | 0              | null          | 0               | 0            | null                 | null      | null                 | 0              | Running       | null   | 0        | 0              |
```

Comment:
I cannot see the type before (see above remark) so I do know it is valid, but anyway, shouldn't all tables just be "unchanged" for this scenario?

Reply:
fixed